CDP4Reporting/Parameters/ReportingParameter.cs

```diff
@@ -112,7 +118,32 @@ public ReportingParameter(string name, Type type, object defaultValue, string fi
             this.Name = name;
             this.Type = type;
             this.DefaultValue = defaultValue;
+
+            this.ForceDefaultValue = defaultValue != null && this.GetDefault(type) != defaultValue;
+
             this.FilterExpression = filterExpression;
         }
+
+        /// <summary>
+        /// Get the default value of a specific <see cref="Type"/>
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/></param>
+        /// <returns>The default value</returns>
+        private object GetDefault(Type type)
+        {
+            return this.GetType()
+                .GetMethod("GetDefaultGeneric", BindingFlags.NonPublic | BindingFlags.Instance)?
```

Comment:
* it would be better to use `nameof(...)` to get the method name instead of hardcoding it
* the "?" for the null-conditional operator should not be separated from the "." on the next line (it should be on the next line, e.g. `?.MakeGeneric...`)